src/main/java/com/sprint/mission/discodeit/service/basic/BasicMessageService.java

```diff
@@ -1,89 +1,137 @@
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
 
+import java.time.Instant;
 import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
+@Service
+@RequiredArgsConstructor
 public class BasicMessageService implements MessageService {
     private final MessageRepository messageRepository;
     private final UserRepository userRepository;
     private final ChannelRepository channelRepository;
-
-    public BasicMessageService(MessageRepository messageRepository, UserRepository userRepository, ChannelRepository channelRepository) {
-        this.userRepository = userRepository;
-        this.messageRepository = messageRepository;
-        this.channelRepository = channelRepository;
-    }
+    private final BinaryContentRepository binaryContentRepository;
+    private final UserStatusService userStatusService;
 
     @Override
-    public Message create(String content, User writer, Channel channel) {
-        if (content.isEmpty()) {
+    public Message create(CreateMessageRequest request) {
+        if (request.content().isEmpty()) {
             throw new ServiceException(ErrorCode.EMPTY_CONTENT);
         }
 
-        existUser(writer);
-        existChannel(channel);
-        Message message = new Message(content, writer, channel);
+        // 작성자와 채널에 대한 검증
+        validUser(request.writerID());
+        validChannel(request.channelID());
+
+        // User가 메시지를 만들면 UserStatus 업데이트하기
+        UserStatus userStatus = userStatusService.findByUserId(request.writerID()).orElseThrow(() -> new ServiceException(ErrorCode.CANNOT_FOUND_USERSTATUS));
+        UpdateUserStatusRequest updateRequest = new UpdateUserStatusRequest(userStatus.getId(), Instant.now());
+        userStatusService.update(updateRequest);
+
+        // 해당 첨부자료가 레포지토리에 저장되어 있는지 확인
+        if (request.attachmentsID() != null) {
+            for (UUID attachmentID : request.attachmentsID()) {
+                Optional<BinaryContent> attachment = binaryContentRepository.findById(attachmentID);
+                if (attachment.isEmpty()) {
+                    throw new ServiceException(ErrorCode.CANNOT_FOUND_ATTACHMENT);
+                }
+            }
+        }
+
+
+        Message message = new Message(request.content(), request.writerID(), request.channelID(), request.attachmentsID());
         messageRepository.save(message);
         return message;
     }
 
     @Override
-    public List<Message> getMessageByUser(User writer) {
-        existUser(writer);
-        return messageRepository.findByUser(writer);
+    public List<Message> findAllByChannelId(UUID channelID) {
+        validChannel(channelID);
+
+        List<Message> messages = messageRepository.findAll();
+        return messages.stream()
+                .filter(message -> message.getChannelID().equals(channelID))
+                .collect(Collectors.toList());
+
```

Comment:
```suggestion
    public List<Message> findAllByChannelId(UUID channelID) {
        validChannel(channelID);
        return messageRepository.findAllByChannelId(channelId);
```
선택적으로 가져오는건 어때요? 데이터의 개수가 많을수록 인메모리에 들고 있는건 부담이에요!